### ISSUE TITLE
GCP Pub/Sub source periodically saying it got invalid argument

### ISSUE THREAD

**Issue by gschier** [vectordotdev/vector#13075]
### Problem

We at [Railway](railway.app) tried out the the `gcp_pubsub` source in production today. Apart from some serious [performance issues](https://github.com/vectordotdev/vector/issues/12846), we also found some bugs. One of which is an `InvalidArgument` error. Here's what it looks like in our logs:

```text
ERROR source{component_kind="source" component_id=in_pubsub component_type=gcp_pubsub component_name=in_pubsub}: vector::internal_events::gcp_pubsub: Failed to fetch events. error=status: InvalidArgument, message: "Request contains an invalid argument.", details: [], metadata: MetadataMap { headers: {} } error_code="failed_fetching_events" error_type="request_failed" stage="receiving"
```

### Configuration

```text
data_dir = "${LOGGER_VECTOR_PATH}"

[api]
enabled = true
playground = false

#[sources.in_pubsub]
#type = "gcp_pubsub"
#project = "railway-infra"
#subscription = "railway-production-logs-sub"

#[transforms.extract_pubsub_message]
#type = "remap"
#inputs = [ "in_pubsub" ]
#source = ". = parse_json!(.message)"

[sinks.out_file]
type = "file"
inputs = ["extract_pubsub_message"]
compression = "none"
encoding = "ndjson"
path = "${LOGGER_VECTOR_PATH}/out_file/logs/%F_%H/{{ appname }}.log"

[sinks.out_gcp_storage]
type = "gcp_cloud_storage"
inputs = ["extract_pubsub_message"]
bucket = "${LOGGER_BUCKET_NAME:-railway-develop-logs}"
key_prefix = "logs/{{ appname }}/"
compression = "none"
buffer.type = "disk"
buffer.max_size = 1_500_000_000_000 # 1.5 TB
batch.timeout_secs = 600 # 10 Minutes
batch.max_bytes = 10_000_000 # 1 MB
filename_append_uuid = false
encoding = "ndjson"
# Note: Time in `filename_time_format` uses time-of-last-log unlike `key_prefix` which uses time-of-first-log
filename_time_format = "%F/%T"
```

### Version

vector 0.22.0 (x86_64-unknown-linux-gnu 5e937e3 2022-06-01)

### References

#12846

**Comment by jszwedko** [comment]
I think this is a duplicate of https://github.com/vectordotdev/vector/issues/12660 . We are planning to cut a release with the fix for this shortly (likely tomorrow).

**Comment by jszwedko** [comment]
Thanks for the report though!

**Comment by jszwedko** [comment]
Ah, nvm, you hit this in 0.22.0. I'll reopen for investigation.

**Comment by gschier** [comment]
Ah yep, looks to be the same issue. My bad, I totally forgot to do a search before posting!

**Comment by jszwedko** [comment]
Hey @gschier ! We released 0.22.1. Could you try that and let us know if you still see this issue?

**Comment by gschier** [comment]
Unfortunately we can't try until #12846 is worked on, as we've only noticed the issue in our production environment but I'm looking forward to testing it out when possible!
